### PR TITLE
Text sizescale

### LINKED ISSUE #248: More granular text DPI scale
The current scale of small, default, large and XL is not enough for most displays, and even I find myself using dconf to change my text DPI 100% of the time.
I think that an iOS styled sanity-checked scale would be less overkill compared to users using 'Large' when they realistically only need 5% adjustment.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/94184737-more-granular-text-dpi-scale?utm_campaign=plugin&utm_content=tracker%2F57462545&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F57462545&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #289: Text size presets are either too small or too large
## Problem

I installed OS 6 on my laptop and using an external monitor with it. But the text sizes that are available on the system settings are either too small or too large for me. Can we have a slider for text sizes where we can comfortably choose which is the optimal point for us?

### PR DESCRIPTION
**Pull request by danrabbit**
- [x] Based on #297

Fixes #289
Fixes #248 

Screenshot from 2021-11-10 12 45 08

### CODE REVIEW

File: src/Views/Text.vala [hunk 1]
```diff
@@ -70,52 +70,51 @@ public class PantheonShell.Text : Gtk.Grid {
...
-
-        interface_settings.changed.connect (() => {
-            text_size_modebutton.set_active (get_text_scale (interface_settings));
-        });
+        interface_settings.bind ("text-scaling-factor", text_size_adjustment, "value", SettingsBindFlags.GET);
 
-        text_size_modebutton.mode_changed.connect (() => {
-            interface_settings.set_double (TEXT_SIZE_KEY, TEXT_SCALE[text_size_modebutton.selected]);
+        // Setting scale is slow, so we wait to keep UI responsive
+        text_size_scale.button_release_event.connect (() => {
```
Comment (cassidyjames) [inline src/Views/Text.vala:82]: So the problem with this approach is that it's not accessible with the keyboard. Gotta keep that. :smile:

File: src/Views/Text.vala [hunk 2]
```diff
@@ -30,20 +28,22 @@ public class PantheonShell.Text : Gtk.Grid {
...
+        var text_size_adjustment = new Gtk.Adjustment (-1, 0.75, 1.5, 0.05, 0.05, 0.05);
+
+        var text_size_scale = new Gtk.Scale (Gtk.Orientation.HORIZONTAL, text_size_adjustment) {
+            draw_value = false
+        };
+        text_size_scale.add_mark (0.75, Gtk.PositionType.BOTTOM, _("Small"));
+        text_size_scale.add_mark (1, Gtk.PositionType.BOTTOM, _("Default"));
+        text_size_scale.add_mark (1.25, Gtk.PositionType.BOTTOM, null);
+        // Avoid rendering bug when putting mark at the very end
+        text_size_scale.add_mark (1.45, Gtk.PositionType.BOTTOM, _("Largest"));
```
Comment (cassidyjames) [inline src/Views/Text.vala:46]: This makes me wonder if we need the small and largest marks at all; they're kind of self-evident based on the design, and we would avoid this funkiness.

**Review by cassidyjames — CHANGES_REQUESTED** [review]
Need to fix keyboard interactivity, I think I messed up resolving merge conflicts (and will try to fix that), and I have a question about the design:

Should we just go with the scale + spinbutton like we do in both Keyboard and Mouse & Touchpad settings? It seems like this is an appropriate place and would make it easier if you know a specific text scale works well on your machine versus having to just guess based on the current design. I also think it would let us address my other inline comment about the start and end marks, since the scale would be even more self-evident.

**Comment by danrabbit** [comment]
Yeah good call about the spin button

**Comment by danrabbit** [comment]
@cassidyjames thanks for the thorough review here! I changed things to match the design of scale + spinbutton we have in other views like Mouse & Touchpad. So, removed the labels, notches on top. I also changed the set method from button release on the scale to a timeout on the adjustment. This fixes the keyboard case and should still have the same effect of keeping the UI from doing a stuttering resize while you're still dragging

**Review by cassidyjames — APPROVED** [review]
Nice work, this seems a lot better.

**Pull request merged**